posts/draft.md

```diff
@@ -0,0 +1,174 @@
+multiple flavors of Markdown as Nikola plugins. Additionally, Nikola is probably unique
+in the Python static site generators arena because it is supporting Jupyter Notebook
+from the core package!
+
+So, all the pieces were asking me to explore this space and I could not resist ;-)
+
+# First impressions (someone has beaten me to it...)
+
+Ok, the first idea coming to my mind was to somehow implement MyST support in Nikola.
+But, hey, somebody probably already thought about this before me. So I made a search to
+see if this was not previously explored by others. And indeed that was the case!
+Roberto, the Nikola creator, wrote a plugin for MyST about a year ago (there is a brief
+blog post talking about it [here](https://ralsina.me/tr/es/weblog/posts/new-plugins-for-nikola.html)).
+That's great, isn't it? Obviously, I quickly jumped into looking at the codebase for that
+plugin and the core of the implementation was quite simple actually:
+
+```python
+output = myst_parser.main.to_html(new_data)
+```
+
+As you can see, it is just a matter to use the Python MyST API to generate the HTML
+output the Nikola machinery is going to write to build the blog post.
+
+Problem solved! Someone else had the idea, implemented it and now I can use it, right?
+Well... actually it was not that simple...
+
+# Current limitations (there is always a but...)
+
+One of the key concepts in the MyST (and rST) world is the support for [roles](https://myst-parser.readthedocs.io/en/latest/sphinx/intro.html#reference-a-section-label-with-a-role) and [directives](https://myst-parser.readthedocs.io/en/latest/sphinx/intro.html#extend-markdown-with-a-directive). People coming from the rST world would be
+tempted to use MyST and the MyST plugin for Nikola if they have support for the main
+pieces they use on a daily basis to save some time, right?
+
+So what happens when you try to use the MyST Python API to render a simple directive:
+
+```ipython
+In [1]: from myst_parser.main import to_html
+
+In [2]: text = """
+   ...: ```{admonition} This is my admonition
+   ...: This is my note
+   ...: ```
+   ...: """
+
+In [3]: to_html(text)
+Out[3]: '<pre><code class="language-{admonition}">This is my note\n</code></pre>\n'
+```
+
+Mmm... that HTML code does not look like an admonition!
+When I quickly search into the MyST and Nikola issue trackers, I quickly found some
+[report](https://github.com/executablebooks/MyST-Parser/issues/307) about this problem.
+
+One of the answers from a core contributor was particularly interesting...
+
+> > I don't know if it's intended to behave this way.
+
+> Heya, yes it is; you have to use sphinx to fully parse to HTML
+
+Wait? Why? I was expecting the MyST Python API to produce the expected HTML for roles
+and directives and that is not supported? Is the API incomplete? Is this intended by
+design? A lot of questions came to my mind and, as usual, the answer was in the
+documentation and the codebase.
```

Comment:
Just a note that this might come across as kind-of negative on the myst parser...it might be better to just get straight to the point, e.g. "This surprised me, as I assumed that roles and directives would be supported by the Python API, but I learned that it is more complicated than that."

Reply:
Thank you... the negativity you mentioned did not pass through my head at any time. I will do the change as requested.